framework/source/class/qx/event/handler/Focus.js

```diff
@@ -1056,11 +1056,12 @@ qx.Class.define("qx.event.handler.Focus",
           return focusedElement;
         }
         // Check compound widgets
-        var widget = qx.ui.core.Widget.getWidgetByElement(focusedElement),
-          textField = widget && widget.getChildControl && widget.getChildControl("textfield", true);
-
-        if (textField) {
-          return textField.getContentElement().getDomElement();
+        var widget = qx.ui.core.Widget.getWidgetByElement(focusedElement);
+        if(widget) {
+          var textField = widget.getChildControl && widget.hasChildControl("textfield") && widget.getChildControl("textfield", true);
```

Comment:
as we're sniffing by method name, then it's necessary to sniff for `hasChildControl` also - there is no contract that says that `hasChildControl` will exist if `getChildControl` does.

Reply:
Hmm. As of the api docs `qx.ui.core.Widget.getWidgetByElement` allways returns an instance of `qx.ui.core.Widget` which always should implement `hasChildControl` and `getChildControl`.
I think we don't need the checks at all.